server/routes/referrals/referralsController.ts

```diff
@@ -39,15 +45,67 @@ export default class ReferralsController {
   }
 
   async createReferral(req: Request, res: Response): Promise<void> {
-    const referral = await this.interventionsService.createDraftReferral(res.locals.user.token)
+    let error: FormValidationError | null = null
+
+    const crn = req.body['service-user-crn']
+    if (crn === null || crn === '') {
+      error = {
+        errors: [
+          {
+            formFields: ['service-user-crn'],
+            errorSummaryLinkedField: 'service-user-crn',
+            message: errorMessages.startReferral.crnEmpty,
+          },
+        ],
+      }
+    } else {
+      try {
+        await this.communityApiService.getOffenderByCRN(crn)
+      } catch (e) {
+        if (e.status === 404) {
+          error = {
+            errors: [
+              {
+                formFields: ['service-user-crn'],
+                errorSummaryLinkedField: 'service-user-crn',
+                message: errorMessages.startReferral.crnNotFound,
+              },
+            ],
+          }
+        } else {
+          logger.error('crn lookup failed', e)
+          error = {
+            errors: [
+              {
+                formFields: ['service-user-crn'],
+                errorSummaryLinkedField: 'service-user-crn',
+                message: errorMessages.startReferral.unknownError,
+              },
+            ],
+          }
+        }
+      }
+    }
+
+    if (error === null) {
+      const referral = await this.interventionsService.createDraftReferral(res.locals.user.token, crn)
 
-    // fixme: this sets some static data for the new referral which will need to be
-    //  changed to allow these fields to be set properly
-    await this.interventionsService.patchDraftReferral(res.locals.user.token, referral.id, {
-      serviceCategoryId: '428ee70f-3001-4399-95a6-ad25eaaede16',
-    })
+      // fixme: this sets some static data for the new referral which will need to be
+      //  changed to allow these fields to be set properly
+      await this.interventionsService.patchDraftReferral(res.locals.user.token, referral.id, {
+        serviceCategoryId: '428ee70f-3001-4399-95a6-ad25eaaede16',
+      })
 
-    res.redirect(303, `/referrals/${referral.id}/form`)
+      res.redirect(303, `/referrals/${referral.id}/form`)
+    } else {
+      // this isn't DRY, but is consistent with existing controller code
```

Comment:
is there a better way to do this?

Reply:
I _think_ we can call `this.startReferral(req, res)` here - I feel like we might be able to do that elsewhere in the `referralsController` too and dry it up?

Reply:
oh no, we'd need to pass it an error. Leaving as-is for now.